pkg/sensors/accelerator/device/device.go

```diff
@@ -0,0 +1,156 @@
+package device
+
+import (
+	"sync"
+	"time"
+
+	"golang.org/x/exp/maps"
+	"k8s.io/klog/v2"
+)
+
+const (
+	DUMMY = iota
+	HABANA
+	DCGM
+	NVML
+)
+
+var (
+	globalRegistry *DeviceRegistry
+	once           sync.Once
+)
+
+type DeviceType int
+
+// Function prototype to create a new deviceCollector.
+type deviceStartupFunc func() DeviceInterface
+type DeviceRegistry struct {
+	gpuDevices   map[DeviceType]deviceStartupFunc // Static map of supported gpuDevices.
+	dummyDevices map[DeviceType]deviceStartupFunc // Static map of supported dummyDevices.
+}
+type DeviceInterface interface {
+	// Name returns the name of the device
+	Name() string
+	// DevType returns the type of the device (nvml, dcgm, habana ...)
+	DevType() DeviceType
+	// DevTypeName returns the type of the device (nvml, dcgm, habana ...) as a string
+	DevTypeName() string
+	// GetHwType returns the type of hw the device is (gpu, processor)
+	HwType() string
+	// InitLib the external library loading, if any.
+	InitLib() error
+	// Init initizalizes and start the metric device
+	Init() error
+	// Shutdown stops the metric device
+	Shutdown() bool
+	// DevicesByID returns a map with devices identifying then by id
+	DevicesByID() map[int]any
+	// DevicesByName returns a map with devices identifying then by name
+	DevicesByName() map[string]any
```

Comment:
Perhaps we need a better name for `DeviceInterface` since this seems to suggest that a `device` has `DevicesByName()` that a `Device` needs to implement. 

I also wonder if the implementation would change between Devices, which if you look at dcgm is 
```

func (d *GPUDcgm) DevicesByName() map[string]any {
	devices := make(map[string]any)
	return devices
}
```
For Habana is 
```
func (g *GPUHabana) DevicesByName() map[string]any {
	devices := make(map[string]interface{})
	return devices
}

```
Is this used?